cvmfs/catalog_mgr_rw.cc

```diff
@@ -837,6 +847,34 @@ shash::Any WritableCatalogManager::SnapshotCatalog(WritableCatalog *catalog)
   return hash_catalog;
 }
 
+void WritableCatalogManager::Balance() {
+  CatalogList catalog_list = GetCatalogs();
```

Comment:
It is a protected method, that's why I left it that way. In any case it returns a reference to the attribute, which could be anyway modified after some time in other parts of the code by other threads